### ISSUE TITLE
distance method is not used in GraphCoords

### ISSUE THREAD

**Issue by SamGG** [exaexa/EmbedSOM#2]
Hi,

I noticed that dist.method is passed but not used in GraphCoords (code below).

https://github.com/exaexa/EmbedSOM/blob/d8476c145f6e60c56aa899d987096e70dcdbd57b/R/coords.R#L208

Moreover, it is recommended to exponentiate the distance, but I don't see a parameter/way to achieve this. Could you elaborate a way to do it?

https://github.com/exaexa/EmbedSOM/blob/d8476c145f6e60c56aa899d987096e70dcdbd57b/R/coords.R#L213-L223

**Comment by SamGG** [comment]
I guess

https://github.com/exaexa/EmbedSOM/blob/d8476c145f6e60c56aa899d987096e70dcdbd57b/R/coords.R#L220

should be replaced by

```
distFn(as.matrix(stats::dist(map$codes, method=dist.method))))))
```

and 

https://github.com/exaexa/EmbedSOM/blob/d8476c145f6e60c56aa899d987096e70dcdbd57b/R/coords.R#L205

should be explicited by

```
#' (e.g. by exponentiating them slightly using distFn=function(x)x^1.5).
```

I set 1.5, but what range of values do you consider to be tested?

Best.

**Comment by exaexa** [comment]
Hi!

thanks for spotting this, looks like I neglected this when unifying the function interfaces. If you have a bit of time to make your suggested changes into a pull request please do, they look OK to me and it will speed up stuff a bit. I'll get to fixing this myself most likely sometime next week.

On a side note, I got a request from CRAN automated checking sometime last week to add some other changes to graphy coords, might be the case these are a similar issue.

Thanks again!
-mk

**Comment by SamGG** [comment]
Hi,
I am also working on removing the dependency on FNN and allowing the dist.method for the KNN. Should I put it on a different PR?
Thanks for your feedback.

**Comment by exaexa** [comment]
> Hi, I am also working on removing the dependency on FNN and allowing the dist.method for the KNN. Should I put it on a different PR?

Yes, if we can get rid of a dependency this is veeeeeery welcome. If you have code, please do in a separate PR though. (Or do in the same one but I'll have to split them :D )

**Comment by SamGG** [comment]
Currently, GraphCoords uses no initialization on the contrary to others methods that call a layout.
It is possible (or not) to either naively mimic MSTCoords or UMatrixCoords
```
      coords = if(is.null(map$grid)) NULL else as.matrix(map$grid),
```
or to go for a more elaborted approach that does not change the current behavior (default coords=NULL) but uses map$grid when coords=TRUE or uses a given matrix of coordinates
```
    if (is.null(coords)) {
    } else if (is.logical(coords) && coords == TRUE) {
      coords <- if (is.null(map$grid)) NULL else as.matrix(map$grid)
    } else if (is.matrix(coords)) {
      stopifnot(ncol(coords) == dim, nrow(coords) == nrow(map$codes))
    }
```
Similarly, kNNCoords uses no initialization.
It's up to you ;-)

**Comment by exaexa** [comment]
This looks like you might have hit the same issue as CRAN was reporting (`coords` are getting into C code somewhere as NULL, which will apparently become a problem in next R versions).

Anyway yeah, looks like more inconsistency, partially caused also by the FNN's euclidean-only limitation. I'll try to unify this a bit. (The code is from ~2020, maybe there are better packages now).

**Comment by SamGG** [comment]
The error reported by CRAN concerns RandomMap and only on platform IIUC. This is different from what I pointed above that could not be catched by CRAN checks IMHO.
A simple apply is able to find the nearest neighbors for 20*20 points. I'll send the PR.

**Comment by exaexa** [comment]
@SamGG  I assume this is completely solved by the merged #4  , right?

**Comment by SamGG** [comment]
Hi,
the first is solved by the merge. The part concerning the coordinates is not in the PR IIRC. I didn't look in the eror reported by the CRAN.
I close the issue and will reopen when I have time to look at it.
Best.

**Comment by exaexa** [comment]
Yes that one is different, I'll handle it elsewhere. Thanks!
